test/Chainweb/Test/Utils.hs

```diff
@@ -992,19 +999,22 @@ runTestNodes
     -> RocksDb
     -> logger
     -> ChainwebVersion
-    -> Natural
+    -> Word
     -> MVar (PeerInfo, Port)
+    -> Map Word (FilePath, FilePath)
+       -- ^ A Map from Node Id to (Pact DB Dir, RocksDB Dir).
     -> IO ()
-runTestNodes testLabel rdb logger ver n portMVar =
-    forConcurrently_ [0 .. int n - 1] $ \i -> do
-        threadDelay (1000 * int i)
-        let baseConf = config ver n
-        conf <- if
-            | i == 0 ->
-                return $ bootstrapConfig baseConf
-            | otherwise ->
-                setBootstrapPeerInfo <$> (fst <$> readMVar portMVar) <*> pure baseConf
-        node testLabel rdb logger portMVar conf i
+runTestNodes testLabel rdb logger ver n portMVar dbDirs = do
+    let nids = [0 .. n - 1]
+    let nidWithDirs = mapMaybe (\nid -> (nid,) <$> Map.lookup nid dbDirs) nids
```

Comment:
Throw an error if there is a missing dbDir instead of using `mapMaybe`, we want to be sure that we're running as many nodes as we requested. Probably better even, you could instead pass in dbDirs as a list and stop passing in `n`, with `n` implicitly set to `length dbDirs`.